docker-compose.yml

```diff
@@ -0,0 +1,53 @@
+version: "2"
+
+services:
+  pda:
+    image: parasoft/demo-app:${PDA_IMAGE_TAG:-main}
+    ports:
+      - "8080:8080"
+      - "9001:9001"
+      - "61623:61623"
+      - "61624:61624"
+      - "61626:61626"
+    working_dir: /usr/local/tomcat/data
+    volumes:
+      - "pda_data:/usr/local/tomcat/data"
+      - "./lib/ParasoftJDBCDriver.jar:/usr/local/tomcat/webapps/ROOT/WEB-INF/lib/ParasoftJDBCDriver.jar:ro"
+      - "./config/application.properties:/usr/local/tomcat/data/config/application.properties:ro"
+    environment:
+      - JAVA_OPTS=-Dspring.kafka.bootstrap-servers=kafka:9092
+    depends_on:
+      - kafka
+  kafka:
+    # https://github.com/bitnami/containers/blob/d559d7c399eec5f0249806fe03981291df013af6/bitnami/kafka/README.md
+    image: docker.io/bitnami/kafka:3.3
+    ports:
+      - "9093:9093"
+    volumes:
+      - "kafka_data:/bitnami"
+    environment:
+      - KAFKA_CFG_ZOOKEEPER_CONNECT=zookeeper:2181
+      - ALLOW_PLAINTEXT_LISTENER=yes
+      - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=INTERNAL:PLAINTEXT,EXTERNAL:PLAINTEXT
+      - KAFKA_CFG_LISTENERS=INTERNAL://:9092,EXTERNAL://:9093
+      - KAFKA_CFG_ADVERTISED_LISTENERS=INTERNAL://kafka:9092,EXTERNAL://${EXTERNAL_HOSTNAME:-localhost}:9093
```

Comment:
Should we also expose the port for configuring external server?

Reply:
Also, can we connect to external Kafka server when only running a pda container? (like using application.properties). I think the docker-compose should be only used when we use Kafka in docker.

Reply:
> Should we also expose the port for configuring external server?

The `EXTERNAL` here is a Kafka listener in the Kafka server(Docker) used for connection from outside of this Docker network. It's not external Kafka server.

See: [Kafka Listeners - Explained](https://rmoff.net/2018/08/02/kafka-listeners-explained/)

Reply:
> Also, can we connect to external Kafka server when only running a pda container? (like using application.properties). I think the docker-compose should be only used when we use Kafka in docker.

Yes, there's no need to use Docker Compose if we don't need the Kafka in Docker. We can configure the Kafka server, and specify the Kafka server in PDA `application.properties` config file and start PDA container with a `docker run` command as usual.